app/models/manageiq/providers/openshift/container_manager/refresher.rb

```diff
@@ -3,9 +3,24 @@ module Openshift
     class ContainerManager::Refresher < ManageIQ::Providers::BaseManager::Refresher
       include ::EmsRefresh::Refreshers::EmsRefresherMixin
 
+      def self.entities
+        %w(routes projects)
+      end
+
       def parse_inventory(ems, _targets = nil)
-        all_entities = ems.with_provider_connection(&:all_entities)
-        all_entities.merge!(ems.with_provider_connection(:service => ManageIQ::Providers::Kubernetes::ContainerManager.ems_type, &:all_entities))
+        all_entities = ems.with_provider_connection { |client| fetch_entities(client, self.class.entities) }
+        fetch = proc do |client|
+          fetch_entities(
+            client,
+            ManageIQ::Providers::Kubernetes::ContainerManager::Refresher.entities
+          )
+        end
+        all_entities.merge!(
```

Comment:
openshift_entities = ...
    kubernetes_entities = ...
    ...
    ContainerManager::RefreshParser.ems_inv_to_hashes(openshift_entities + kubernetes_entitis)

Which seems to express in a more clear way what we collect and what we return.